### PR TITLE
import trigger release logic from kotlin-mpp-godtools-shared build workflow

### PR DESCRIPTION
**Pull request by frett**
copied the logic added in this PR: https://github.com/CruGlobal/kotlin-mpp-godtools-tool-parser/pull/326

I had to tweak the logic to use the `master` branch instead of `main`

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -197,15 +201,64 @@ jobs:
...
   deploy:
     name: Deploy Library
     runs-on: macos-11
     if: >-
       github.event_name == 'workflow_dispatch' ||
       github.event_name == 'schedule' ||
       github.event_name == 'pull_request' ||
-      (github.event_name == 'push' && (github.ref == 'refs/heads/master' || contains(github.ref, 'refs/tags/')))
-    needs: [build, checkstyle, ktlint, lint, android_tests, ios_tests, js_tests]
+      (github.event_name == 'push' && github.ref == 'refs/heads/master')
```
Comment (wrandall22) [inline .github/workflows/build.yml:260]: No `workflow_dispatch` needed here?

**Review by wrandall22 — COMMENTED** [review]
Unrelated to this PR, but why does this deploy use `macos-11` instead of `ubuntu-latest` like the KMM parser?

File: .github/workflows/build.yml (hunk 1 above)
Comment (frett) [reply .github/workflows/build.yml:260]: it's already in the conditional earlier

**Review by wrandall22 — APPROVED** [review]

**Comment by frett** [comment]
@wrandall22 for KMM to build/deploy any ios binaries/libraries it needs to run in a mac environment.

`gto-support` dependencies are consumed by `godtools-shared`, so it needs to be able to publish the ios binaries.
`godtools-shared` published kmm libraries are only consumed by `godtools-android`, on iOS we use a cocoapod which doesn't rely on the jfrog artifacts.

**Pull request merged**